src/implementations/vanilla/src/basics/form-elements/text-field/tests/tests-text-field.html

```diff
@@ -97,6 +102,33 @@
             IconTextField.setIcon('location');
 
 
+            // TEXT FIELD WITH CLEAR BUTTON
+
+            const ClearButtonHeading = new Hig.Typography({
+                text: "TextField with clear button",
+                type: "h2"
+            });
+
+            const ClearButtonTextField = new TextField({
+                label: "What is your favorite secret society?",
+                placeholder: "Skull & Bones"
+            });
+
+            ClearButtonHeading.mount('.tests-text-field-clear-button-heading');
+            ClearButtonTextField.mount('.tests-text-field-clear-button');
+
+            ClearButtonTextField.onInput(() => {
+              if (ClearButtonTextField.getValue().length === 0) {
```

Comment:
We don't need `getValue` here. The first argument to an event handler like this is the event object.

This will do the trick:

```js
ClearButtonTextField.onInput(event => {
  if (event.target.value.length == 0) {
    ClearButtonTextField.showClearButton();
  } else {
    ClearButtonTextField.hideClearButton();
  }
});
```

Reply:
Good call, thank you

Reply:
Updated in 3c2f090